spec/myenumerables_spec.rb

```diff
@@ -0,0 +1,141 @@
+require './myenumerables.rb'
+
+describe Enumerable do
+  arr = [4, 20, 3, 22]
+  query = 4
+  query_symbol = :+
+
+  describe '.my_each' do
+    it 'Makes self into array unless block is given' do
+      expect(arr.my_each).to be_a(Enumerable)
+    end
+    it 'if block is given' do
+      expect(arr.my_each(&:even?)).to eq(arr.each(&:even?))
+    end
+  end
+
+  describe '.my_each_with_index' do
+    it 'If block is not given' do
+      expect(arr.my_each_with_index).to be_a(Enumerable)
+    end
+    it 'Take an array and gets the element and index of the block_given?' do
+      hash_my_method = {}
+      hash_ruby_method = {}
+      arr.my_each_with_index do |item, index|
+        hash_my_method[item] = index
+      end
+      arr.each_with_index { |item, index| hash_ruby_method[item] = index }
+      expect(hash_my_method).to eq(hash_ruby_method)
+    end
+  end
+
+  describe '.my_select' do
+    it 'Take an array and filters block_given' do
+      expect(arr.my_select).to be_a(Enumerable)
+    end
+    it 'It pushes array when item matches criteria' do
+      result = []
+      arr.each do |element|
+        result << element if element == query
+      end
+      tests = arr.my_select { |element| element == query }
+      expect(tests).to eq(result)
+    end
+  end
+
+  describe '.my_all?' do
+    it 'When there is no block given truthy value' do
+      expect(arr.my_all?).to be_truthy
+    end
+    it 'Take an array and filters to find if an element is included to return true or false' do
+      hash = {}
+      for item in arr do
+        hash[item] = if arr[item] == arr[item.next]
+                       1
+                     else
+                       0
+                     end
+      end
+      result = hash.value?(0)
+      expect(result).to be_truthy
+    end
+  end
+
+  describe '.my_any?' do
+    it 'Takes a block and return true if item in block matches' do
+      expect(arr.my_any?(nil)).to be_truthy
+    end
+    it 'Take an array and filters to find if an element is included to return true or false' do
+      hash = {}
+      for item in arr do
+        hash[item] = if item == query
+                       1
+                     else
+                       0
+                     end
+      end
+      result = hash.value?(1)
+      expect(result).to be_truthy
+    end
+  end
```

Comment:
Let's test when:
- [ ] when a class is passed as an argument
- [ ] when a Regex is passed as an argument

Reply:
Ok added test cases for Classes and RegEx 👍🏽